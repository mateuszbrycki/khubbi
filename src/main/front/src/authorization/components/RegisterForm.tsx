import React, {useEffect} from 'react'
import {useState} from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

export interface RegisterFormProps {
}

export interface RegisterFormActionProps {
    readonly registerUser: (email: string, password: string) => void
}

const RegisterForm: React.FC<RegisterFormProps & RegisterFormActionProps> = (props) => {

    const {registerUser} = props

    const [email, setEmail] = useState<string>("")
    const [password, setPassword] = useState<string>("")
    const [repeatPassword, setRepeatPassword] = useState<string>("")
    const [validationError, setValidationError] = useState<string>("")

    useEffect(() => {
        if (password != repeatPassword) {
            setValidationError("Passwords are not equal");
            return;
        }
        setValidationError("");
    }, [password, repeatPassword]);

    const resetFormValues = () => {
        setEmail("");
        setPassword("");
        setRepeatPassword("");
        setValidationError("");
    }

    return <>
        <h2>Register</h2>
        <Form onSubmit={event => {
            registerUser(email, password);
            event.preventDefault();
            resetFormValues();
        }}>
            <Form.Group className="mb-3" controlId="registerFormEmail">
                <Form.Label>Email address</Form.Label>
                <Form.Control type="text" value={email} placeholder="Enter email"
                              onChange={(event) => setEmail(event.target.value)}
                              required
                />
            </Form.Group>

            <Form.Group className="mb-3" controlId="registerFormPassword">
                <Form.Label>Password</Form.Label>
                <Form.Control type="password" value={password} placeholder="Password"
                              onChange={(event) => setPassword(event.target.value)} required/>
            </Form.Group>

            <Form.Group className="mb-3" controlId="registerFormPasswordRepeat">
                <Form.Label>Repeat Password</Form.Label>
                <Form.Control type="password" value={repeatPassword} placeholder="Repeat Password"
                              onChange={(event) => setRepeatPassword(event.target.value)} required/>
                <Form.Control.Feedback type="invalid">
                    Passwords are not equal.
                </Form.Control.Feedback>
            </Form.Group>
            <Button variant="primary" type="submit">
                Submit
            </Button>
        </Form>

        {validationError}
    </>
}

export default RegisterForm