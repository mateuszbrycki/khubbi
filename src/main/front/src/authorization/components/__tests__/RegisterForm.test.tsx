import RegisterForm from "../RegisterForm";
import ShallowRenderer from "react-test-renderer/shallow";

it('renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<RegisterForm registerUser={(email, password) => null} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});