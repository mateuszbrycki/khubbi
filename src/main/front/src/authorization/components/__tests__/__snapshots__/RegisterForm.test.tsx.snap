// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <h2>
    Register
  </h2>,
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="registerFormEmail"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="registerFormEmail"
        onChange={[Function]}
        placeholder="Enter email"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="registerFormPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="registerFormPassword"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="registerFormPasswordRepeat"
      >
        Repeat Password
      </label>
      <input
        className="form-control"
        id="registerFormPasswordRepeat"
        onChange={[Function]}
        placeholder="Repeat Password"
        required={true}
        type="password"
        value=""
      />
      <div
        className="invalid-feedback"
      >
        Passwords are not equal.
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>,
]
`;
